### PR TITLE
LPS-202597 Preventing the establishment of path parameters without a response API Schema in GET Single element endpoints

### PR DESCRIPTION
**Pull request by jalber786**
**What's new?:**
- Added new validation that prevents the the establishment of path parameters without a response API Schema in GET Single element endpoints.
- Added new test case that check for that new validation.

**What's changed?:**
- Modified some test cases that falied because of the new validation.

See the following **Jira ticket**: [LPS-202597](https://liferay.atlassian.net/browse/LPS-202597)

**NOTE:** The first commit of this PR is  [4bf442f](https://github.com/liferay-headless/liferay-portal/pull/1780/commits/4bf442ff28fe6a62b8c7ca49287ad23882cde9da)

### CODE REVIEW

**Comment by jalber786** [comment]
ci:test:relevant

**Review by sergiojimcos — CHANGES_REQUESTED** [review]
Just a comment

**Comment by jalber786** [comment]
ci:test:sf

**Comment by jalber786** [comment]
ci:test:relevant

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -12894,6 +12894,7 @@ path-parameter-must-be-an-external-reference-code,-id,-or-unique-field=Path para
 path-parameter-property=Path Parameter Property
 path-parameters-are-not-supported-by-get-api-endpoints-with-collection-retrieve-type=Path parameters are not supported by GET API endpoints with "collection" retrieve type.
 path-parameters-are-not-supported-by-post-api-endpoints=Path parameters are not supported by POST API endpoints.
+path-parameters-cannot-be-established-without-a-response-schema-in-use=Path parameters cannot be established without a response schema in use.
```
Comment (sergiojimcos) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:12897]: I prefer something easier in order to read: `Path parameter cannot be set without a response schema.`

**Review by sergiojimcos — CHANGES_REQUESTED** [review]
Just a comment

File: modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APIEndpointRelevantObjectEntryModelListener.java [hunk 2]
```diff
@@ -290,6 +290,17 @@ private void _validateGetAPIEndpoint(
 					(String)values.get("retrieveType")),
 				APIApplication.Endpoint.RetrieveType.SINGLE_ELEMENT)) {
 
+			if (!Validator.isBlank(pathParameter) &&
```
Comment (ccorreagg) [inline modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APIEndpointRelevantObjectEntryModelListener.java:293]: Just a question, is it possible to unset the response schema once the endpoint has been created?

Comment (jalber786) [reply modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APIEndpointRelevantObjectEntryModelListener.java:293]: Yes, I think so but it's currently unavailable through the UI.

**Comment by jalber786** [comment]
ci:test:sf

**Comment by jalber786** [comment]
ci:test:relevant

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 3]
```diff
@@ -12888,13 +12888,13 @@ patch=Patch
 path=Path
 path-and-file-name=Path and File Name
 path-can-not-be-empty=Path cannot be empty.
+path-parameter-cannot-be-set-without-a-response-schema=Path parameter cannot be set without a response schema.
```
Comment (sergiojimcos) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:12891]: Don't forget to do the `buildLanguage`

**Review by sergiojimcos — CHANGES_REQUESTED** [review]
Just a comment

**Comment by jalber786** [comment]
ci:test:relevant

**Comment by jalber786** [comment]
ci:test:sf

**Review by ccorreagg — APPROVED** [review]

**Review by sergiojimcos — APPROVED** [review]
For future PRs, please review the name of the commits like this 97823e6294c60cddb5226f846c78dff88dd395ba, just to not cause misunderstanding

**Comment by jalber786** [comment]
ci:forward

**Comment by jalber786** [comment]
ci:forward:force

**Comment by jalber786** [comment]
I've been analysing locally and the tests are passing. 

It seems that the failures are related to the commits this PR is reverting: https://github.com/brianchandotcom/liferay-portal/pull/146258.

**Pull request closed without merging**